src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -29,12 +29,105 @@
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
 /**
- * Checks the npath complexity against a specified limit (default = 200).
- * The npath metric computes the number of possible execution paths
- * through a function. Similar to the cyclomatic complexity but also
- * takes into account the nesting of conditional statements and
- * multi-part boolean expressions.
+ * <p>
+ * Checks the NPATH complexity against a specified limit.
+ * </p>
+ * <p>
+ * The NPATH metric computes the number of possible execution paths through a
+ * function(method). It takes into account the nesting of conditional statements
+ * and multi-part boolean expressions (A &amp;&amp; B, C || D, E ? F :G and
+ * their combinations).
+ * </p>
+ * <p>
+ * The NPATH metric was designed base on Cyclomatic complexity to avoid problem
+ * of Cyclomatic complexity metric like nesting level within a function(method).
+ * </p>
+ * <p>
+ * Metric was described at <a href="http://dl.acm.org/citation.cfm?id=42379">
+ * "NPATH: a measure of execution pathcomplexity and its applications"</a>.
+ * If you need detailed description of algorithm, please read that article,
+ * it is well written and have number of examples and details.
+ * </p>
+ * <p>
+ * Here is some quotes:
+ * </p>
+ * <blockquote>
+ * An NPATH threshold value of 200 has been established for a function.
+ * The value 200 is based on studies done at AT&amp;T Bell Laboratories [1988 year].
+ * </blockquote>
+ * <blockquote>
+ * Some of the most effective methods of reducing the NPATH value include:
+ * <ul>
+ * <li>
+ * distributing functionality;
+ * </li>
+ * <li>
+ * implementing multiple if statements as a switch statement;
+ * </li>
+ * <li>
+ * creating a separate function for logical expressions with a high count of
+ * variables and (&amp;&amp;) and or (||) operators.
+ * </li>
+ * </ul>
+ * </blockquote>
+ * <blockquote>
+ * Although strategies to reduce the NPATH complexity of functions are important,
+ * care must be taken not to distort the logical clarity of the software by
+ * applying a strategy to reduce the complexity of functions. That is, there is
+ * a point of diminishing return beyond which a further attempt at reduction of
+ * complexity distorts the logical clarity of the system structure.
+ * </blockquote>
+ * <table>
+ * <caption>Examples</caption>
```

Comment:
This is new.
@pbludov Can you explain this addition after approvals?

Reply:
javadoc lint wants caption or summary for each table:

https://stackoverflow.com/questions/22528767/how-to-work-around-the-stricter-java-8-javadoc-when-using-maven